packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -577,22 +578,24 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
      *
      * @param widget widget file resource of which should be revealed and selected
      */
-    async selectWidgetFileNode(widget: Widget | undefined): Promise<void> {
+    async selectWidgetFileNode(widget: Widget | undefined, isSelectedByEditor?: boolean): Promise<void> {
         if (Navigatable.is(widget)) {
             const resourceUri = widget.getResourceUri();
             if (resourceUri) {
                 const { model } = await this.widget;
                 const node = await model.revealFile(resourceUri);
                 if (SelectableTreeNode.is(node)) {
-                    model.selectNode(node);
+                    model.selectNode(node, isSelectedByEditor ? true : false);
```

Comment:
No need for the ternary:

```suggestion
                    model.selectNode(node, isSelectedByEditor);
```

Reply:
Or `!!isSelectedByEditor` to be certain it's a boolean, as you have below on l. 598.

Reply:
> Or `!!isSelectedByEditor` to be certain it's a boolean, as you have below on l. 598.

done.